rust/catalyst-types/src/catalyst_id/mod.rs

```diff
@@ -676,6 +676,12 @@ impl TryFrom<&[u8]> for CatalystId {
     }
 }
 
+impl From<&CatalystId> for Vec<u8> {
```

Comment:
What is this for?
IF this is so we can serialize a Vec<u8> as cbor, then its wrong.
We should have serialize methods on CatalystId, not sidestep the type system.